test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Smart/SmartSearchTests.cs

```diff
@@ -0,0 +1,358 @@
+        }
+
+        [SkippableFact]
+        public async Task GivenScopesWithReadForObservation_WhenChainedSearchWithPatientName_ThrowsInvalidSearchException()
+        {
+            Skip.If(
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4 &&
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4B,
+                "This test is only valid for R4 and R4B");
+
+            // try to query both the patient and the observation using revinclude
+            var query = new List<Tuple<string, string>>();
+            query.Add(new Tuple<string, string>("subject:Patient.name", "SMARTGivenName1"));
+
+            var scopeRestriction = new ScopeRestriction("Observation", Core.Features.Security.DataActions.Read, "patient");
+
+            ConfigureFhirRequestContext(_contextAccessor, new List<ScopeRestriction>() { scopeRestriction });
+
+            await Assert.ThrowsAsync<InvalidSearchOperationException>(() =>
+                _searchService.Value.SearchAsync("Observation", query, CancellationToken.None));
+        }
+
+        [SkippableFact]
+        public async Task GivenScopesWithReadForObservationAndPatient_WhenChainedSearchWithPatientName_ThenObservationResourceReturned()
+        {
+            Skip.If(
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4 &&
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4B,
+                "This test is only valid for R4 and R4B");
+
+            // try to query both the patient and the observation using revinclude
+            var query = new List<Tuple<string, string>>();
+            query.Add(new Tuple<string, string>("subject:Patient.name", "SMARTGivenName1"));
+
+            var scopeRestriction = new ScopeRestriction("Observation", Core.Features.Security.DataActions.Read, "patient");
+            var scopeRestriction2 = new ScopeRestriction("Patient", Core.Features.Security.DataActions.Read, "patient");
+
+            ConfigureFhirRequestContext(_contextAccessor, new List<ScopeRestriction>() { scopeRestriction, scopeRestriction2 });
+
+            var result = await _searchService.Value.SearchAsync("Observation", query, CancellationToken.None);
+
+            Assert.Collection(
+                result.Results,
+                r => Assert.Equal("smart-observation-A1", r.Resource.ResourceId),
+                r2 => Assert.Equal("smart-observation-A2", r2.Resource.ResourceId));
+        }
+
+        [SkippableFact]
+        public async Task GivenScopesWithReadForPatient_WhenRevChainedSearchWithObservationCode_ThrowsInvalidSearchException()
+        {
+            Skip.If(
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4 &&
+                ModelInfoProvider.Instance.Version != FhirSpecification.R4B,
+                "This test is only valid for R4 and R4B");
+
+            // try to query both the patient and the observation using revinclude
```

Comment:
This might be my lack of knowledge in revinclude, but isn't this block (line 229 and 230) only querying observation using revinclude? Asking because the comment says `try to query *both the patient and the observation* using revinclude`

Reply:
Correct!  The comment was a cut and paste :).  It is only querying for patient resources, and it is trying to find the Patient resources where they have an Observation resource referring to them which also has a code of 4548-4.  Comments are updated.